runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -5183,6 +5183,15 @@ bool TR_J9InlinerPolicy::isJSR292SmallHelperMethod(TR_ResolvedMethod *resolvedMe
       case TR::java_lang_invoke_ConvertHandleFilterHelpers_number2J:
       case TR::java_lang_invoke_MethodHandle_doCustomizationLogic:
       case TR::java_lang_invoke_MethodHandle_undoCustomizationLogic:
+      case TR::java_lang_invoke_MethodHandleImpl_ArrayAccessor_setElementI:
```

Comment:
While I understand that inlining these small methods is a good thing in likely all cases, I was under the impression that there was some other "argument type mismatch" reason for why these methods were not getting inlining using the normal heuristics (and that Devin was working to fix those heuristics). If this is somehow incorrect, please clarify that, or confirm that this is a workaround fix (that does seem to have precedent for other MH inlining that was driven by the same routine in the inliner) until that more general problem is fixed.

Reply:
I found two separate issues preventing these methods getting inlined, which are:
* The "argument type mismatch" resulting from the method args in the signature string containing arrays (`[`), that @jdmpapin's patch addresses and I expect will be contributed separately. The benchmark results I shared with you earlier today did not include Devin's patch in the build.
* These methods being added to trimmed list of callees when we run out of inliner budget for very complex compilations, despite them being on hot paths and being very simple.

Both of these issues need to be addressed to reliably inline these simple methods. This PR addresses the latter issue for these ArrayAccessor methods, so that they get picked from the trimmed list of callees because we know that they are simple and they are going to be beneficial in all cases. Unfortunately we do not currently have any general heuristics that would enable us to recognize simple methods like these that we should inline under all circumstances, which would enable us to shorten the growing list of methods we have in `alwaysWorthInlining` or `isJSR292SmallHelperMethod`, etc.

Reply:
The spurious type incompatibility should be fixed by #17168